### ISSUE TITLE
Question: what's about unloaded pages?

### ISSUE THREAD

**Issue by OhSoGood** [Lusito/forget-me-not#54]
Hi,

Firefox has the capacity to unload pages to save memory & co - either on start, or while the browser is running (via a native API, that notably the Auto Tab Discard extension and others use - see https://addons.mozilla.org/en-US/firefox/addon/auto-tab-discard/)

Does your extension manage this? That'd be a must.

**Comment by Lusito** [comment]
I haven't actually tested this, but looking at the API documentation, I would assume, that unloaded pages would not affect this extension. I.e. the extension would think tabs are still loaded, keeping the cookies alive.
Since auto-tab-discard uses Firefox's internal methods to do the same thing, it should have the same result.

If you want to help me out, try this with a page that should remove cookies on close (based on your rules) and see if the cookies get removed.

**Comment by Lusito** [comment]
Since cookies should not get deleted in this case, this would be unwanted behavior and I'd have to flag this as a bug. Why would you want those cookies removed? You haven't actually left the domain, so I'd assume that you want to stay logged in.

**Comment by OhSoGood** [comment]
Funny, I would have said the opposite: you keep a page open for a long 
time. In that case, your extension clears its cookies. Page unloading 
simply add another feature, which is to free memory. But from an ageing 
point of view and so about cookies, page unloading should be 
transparent, shouldn't they?

Le 23/04/2018 à 21:25, Lusito a écrit :

**Comment by Lusito** [comment]
Not sure. I see this scenario: You log into a website, go browse on other sites, then go back and notice that you are logged out for no apparent reason. You actually never left the domain, so at the very least, the description of what forget-me-not does is wrong at the moment and needs to be adjusted.

I'm gonna leave this issue open for discussion.

**Comment by B00ze64** [comment]
I agree, if the TAB get's "UnLoaded" somehow but it's still opened, cookies must stay. How does @OhSoGood unload a TAB? Lazy-loaded TABs must also keep their cookies (i.e. restart FF with 100 TABs opened, only 1 will be loaded, cookies for all TABs must stay). Regards,

**Comment by OhSoGood** [comment]
I would have said that to unload page is like to close a page - you just keep the tab open for memory (ie for later use) but everything else should have been torn apart.
That's just my point of view, but from a privacy aspect, for all those keeping many many tabs open for later reading, that means we keep being traced and I would really prefer to have the cookies & co removed.
Should cleaning unloaded page an optional setting in the extension?

**Comment by rmenessec** [comment]
I would definitely prefer that FMN _preserve_ data for unloaded tabs.

Hell, I'd like to see someone write an extension to save and restore form fields for tabs being unloaded, so I don't have to worry about, say, a partially-written comment on GitHub being lost simply because I wasn't on the page for a few minutes, and Auto Tab Discard threw out the tab's memory.

(Auto Tab Discard supposedly no longer discards tabs with filled form data, but apparently it's not perfect, yet...)

**Comment by OhSoGood** [comment]
Indeed, unloading tabs answers to different needs corresponding to 
various situations.

Using a default behavior and, giving other people a simple option to use 
another one, would be maybe the more efficient way - provided it's not 
heavy on the developer side (being a developer myself, I can understand 
that!).

In call cases, thanks for the extension!

Le 27/05/2018 à 19:31, rmenessec a écrit :

**Comment by ElhemEnohpi** [comment]
I'm using Auto Tab Discard, and when a tab gets unloaded its cookies are not deleted by FMN. I would like it to stay that way. It might be nice to have the option, but if there's only one option, it should be the way it is...
